externs/browser/window.js

```diff
@@ -191,9 +191,9 @@ function setImmediate(callback, callbackParams) {}
 function setInterval(callback, opt_delay, callbackParams) {}
 
 /**
- * @param {Function|!TrustedScript|string} callback
+ * @param {function(...?):undefined|!TrustedScript|string} callback
  * @param {number=} opt_delay
- * @param {...*} callbackParams
```

Comment:
Why this loosening?

Reply:
The short answer is that, this is how it appeared in Elemental2 patch. The longer answer is that I was trying to make it consistent with the typing of the parameters passed to the callback function as well as consistent with the other timing related functions (i.e. `setInterval` and `setImmediate`) which are both typed as `{...?}`.

In the context of a parameter accepted by a function defined in an extern I did not think it would have any impact on the code that type checks. Although I have been wrong in the past ;-)

Reply:
Also I believe it has an impact on the code generated in elemental2 ... and moving it to `{...?}` means we could eliminate the patch altogether.

Reply:
Using `*` vs. `?` makes no difference in type checking of calls made to this method.
It will make a difference if you ever try to define a method that is supposed to implement this API.
If you use `*`, then that implementation won't be allowed to access any properties on the `callbackParams` items,
but it will if you use `?`.

`*` is the more correct choice here, since `setTimeout` shouldn't do anything with the `callbackParams` other than pass them to the `callback` function. Picking the wrong one may not break anything now, but could bite someone later on.

Regardng the patch, can't you still eliminate the patch by changing the elemental2 side to expect/define `*` instead of `?`?

Reply:
I moved some code around within the extern (essentially introducing appropriate typedefs) and I believe `*` now works and I also applied this typing to `setInterval`. 

Hopefully that addresses your concerns.

I must admit I never even considered downstream code implementing the code defined by externs ;)